engine/apps/api/serializers/alert_group.py

```diff
@@ -92,10 +95,33 @@ def get_render_for_web(self, obj):
         if not obj.last_alert:
             return {}
 
-        return AlertGroupWebRenderer(obj, obj.last_alert).render()
+        web_templates_modified_at = obj.channel.web_templates_modified_at
+        last_alert_created_at = obj.last_alert.created_at
+
+        CACHE_KEY = f"render_for_web_alert_group_{obj.id}"
+        CACHE_LIFEIME = 60 * 60 * 24
+        cached_render_for_web = cache.get(CACHE_KEY, None)
+
+        # use cache only if cache exists
+        # and cache was created after the last alert created
+        # and either web templates never modified
+        # or cache was created after templates were modified
+        if (
+            cached_render_for_web is not None
+            and cached_render_for_web.get("cache_created_at") > last_alert_created_at
```

Comment:
Is cache_created_at value from cache datetime so you can compare it with last_alert_created_at? I remember, that after putting to and retrieving from cache and became a string.